src/util/util.ts

```diff
@@ -50,11 +50,20 @@ export function forEachReverse<T, U>(arr: readonly T[] | null | undefined, actio
     return undefined;
 }
 
-export function earliestDate(...dates: (Date | undefined)[]) {
-    return dates.reduce((res,d) => (res && d && d < res ? d : res) || d, undefined);
+function defaultCompare<T>(a: T, b: T) {
```

Comment:
Not important, but I prefer a `const defaultCompare = ...` for local functions (but see below).

Reply:
Done :heavy_check_mark: